### PR TITLE
Adding asyncio green mode documentation and a how-to on contributing

### PR DESCRIPTION
**Pull request by LJBD**
This PR contains documentation for asyncio green mode, a how-to on contributing and several other small changes.
It's still a WIP.
It references 1st item on the checklist in #110.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -39,7 +39,7 @@ Python dependencies:
 Build dependencies:
 
 - setuptools_
-- sphinx_
+- sphinx_ < 1.5
```
Comment (AntoineDupre) [inline README.rst]: This is not useful anymore, it has been fixed in PR #112.

File: doc/how-to-contribute.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+Everyone is welcome to contribute to PyTango project.
+If you don't feel comfortable with writing core PyTango we are looking for contributors to documentation or/and tests.
+
+Workflow
+--------
+
+A normal Git workflow is used. You can find how to automate your git branching workflow example_.
+Good practices:
+
+* You can create pull requests, you don't have to use your fork,
```
Comment (AntoineDupre) [inline doc/how-to-contribute.rst]: If someone is not a member, he needs to fork the project to contribute

File: doc/how-to-contribute.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
+.. currentmodule:: tango
```
Comment (AntoineDupre) [inline doc/how-to-contribute.rst:1]: This should be added at the beginning of the how to section

File: doc/green_modes/green_modes_server.rst [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+
+asyncio mode
+~~~~~~~~~~~~
+
+Redirects all user code to an event loop. All user methods become
+coroutines, so you should define them with "async" keyword. DS. code written
+with asyncio. There's no monitor lock!
+
+.. literalinclude:: ../../examples/asyncio_device/asyncio_device_example.py
```
Comment (AntoineDupre) [inline doc/green_modes/green_modes_server.rst]: This path does not exist. 
Should it be ```../../examples/asyncio_green_mode/asyncio_device_example.py``` ?

Comment (LJBD) [reply doc/green_modes/green_modes_server.rst]: Fixed.

**Pull request merged**